app/models/study_room.rb

```diff
@@ -34,6 +34,21 @@ def to_s
     location
   end
 
+  def space_totals
+    @space_totals ||= self.class.group(:general_area).count
+  end
+
+  def space_assigned_totals
+    @space_assigned_totals ||= StudyRoomAssignment.where(released_date: nil).joins(:study_room)
+                                                  .group(:general_area).order(:general_area).count
+  end
+
+  def space_report
+    space_totals.map do |key, total|
+      [key, [total, total, space_assigned_totals[key], total - space_assigned_totals[key]]]
```

Comment:
why there are two 'total' ?

Reply:
should the second total be a different calculation?

Reply:
No, with Study rooms they do not disable them.  That is just a locker construct.